google/cloud/bigquery/table.py

```diff
@@ -1575,12 +1579,51 @@ def __init__(
         self._field_to_index = _helpers._field_to_index_mapping(schema)
         self._page_size = page_size
         self._preserve_order = False
-        self._project = client.project if client is not None else None
         self._schema = schema
         self._selected_fields = selected_fields
         self._table = table
         self._total_rows = total_rows
         self._first_page_response = first_page_response
+        self._location = location
+        self._job_id = job_id
+        self._query_id = query_id
+        self._project = project
+
+    @property
+    def _bqstorage_project(self) -> Optional[str]:
+        """GCP Project ID where BQ Storage API will bill to (if applicable)."""
+        client = self.client
```

Comment:
I wonder what's the difference between `self.client.project` and `self._project`. When the client isn't of storage type, does `self.client.project` still reflect the storage project id?

Reply:
The difference in other contexts is usually described as "billing project" versus "data project". This is easiest to understand in the context of public datasets. I've renamed this private property to reflect that.

I have permission to query tables in the `bigquery-public-data` project and even download the data in bulk with the BigQuery Storage Read API, but I _don't_ have permission to run queries in that project or start a BigQuery Storage Read API session from that project. Instead, I have start the query or bq storage read session in my own project and reference the tables in the other project.

Reply:
Wow I see! so `self._project` is the project for the data/table, and `self.client.project` a.k.a. billing project is the user's own project. So I guess this billing project doesn't just apply to storage?

Reply:
That's true, though for queries we're using `jobs.getQueryResults`, so the project will always be the billing project since that's where temp results for queries are stored.

Reply:
`tabledata.list` is free, but technically we should be overriding the [x-goog-user-project header](https://cloud.google.com/apis/docs/system-parameters#definitions) so that the quota is charged to the billing project. (I don't think we're actually doing that, though)